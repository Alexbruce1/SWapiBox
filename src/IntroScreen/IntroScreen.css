.intro-container {
  background-color: black;
  overflow: hidden;
  height: 100vh;
  width: 100%;
  position: fixed;
}

.title-container {
  background-color: black;
  box-shadow: 0 2rem 2rem black;
  position: relative;
  z-index: 2;
}

.sw-logo {
  color: #FFFF82;
  font-size: 6rem;
  font-family: 'Star Jedi Hollow';
  font-weight: 200;
  line-height: 5rem;
  margin: 0;
  padding: 0;
  position: relative;
  vertical-align: middle;
  z-index: 2;
}

.sw-logo-film {
  /* background-color: green; */
  font-size: 1.8rem;
  font-family: 'Futura Medium' sans-serif;
  font-weight: 400;
  letter-spacing: 8px;
  line-height: 2rem;
  color: #fcfcfc;
  margin: 1rem 0 0 0;
  position: relative;
  text-transform: uppercase;
  z-index: 2;
}

.intro-screen {
  background-color: black;
  height: 100%;
  width: 100%;
}

.intro-container p,
.intro-container div {
  color: #FFFF82;
  margin: auto;
}

#subtitle {
  font-size: 5rem;
  margin-bottom: 2rem;
}

#title {
  font-size: 3rem;
}

.crawl-head,
.crawl-text {
  color: #FFFF82;
  margin: auto;
}

#board {
  /* background-color: green; */
  bottom: 6rem;
  font-family: Century Gothic, sans-serif;
  font-size: 350%;
  font-weight: bold;
  height: 150rem;
  left: calc(50% - 30rem);
  overflow: hidden;
  position: absolute;
  text-align: justify;
  transform: perspective(300px) rotateX(25deg);
  transform-origin: 50% 100%;
  width: 60rem;
}

#board::after {
  bottom: 60%;
  position: absolute;
  top: 0;
}

#content {
  animation: scroll 100s linear 0s;
  margin: auto;
  position: absolute;
  top: 70%;
}

#title, #subtitle {
  text-align: center;
}

@keyframes scroll {
  0% {
    top: 100%;
  }
  100% {
    top: -170%;
  }
}

.intro-logo {
  background-color: black;
  box-shadow: 0 2rem 1rem black;
  left: 0;
  position: absolute;
  padding: 0 35%;
  top: 2.5rem;
  width: 30%;
  z-index: 2;
}

.intro-arrow {
  background-color: transparent;
  border: none;
  bottom: 1rem;
  cursor: pointer;
  display: inline;
  left: calc(50% - 2rem);
  outline: none;
  position: absolute;
}

.rebellion-button {
  height: 4rem;
}

.intro-arrow:hover .rebellion-button {
  filter: invert(50%);
  transition-duration: .4s;
}


.star {
  background-color: white;
  height: 8px;
  position: absolute;
  width: 8px;
}