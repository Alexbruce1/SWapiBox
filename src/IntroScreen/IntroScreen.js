import React from 'react';

import logo from '../images/SWlogo.png'
import rebelClear from '../images/rebel-yellow.svg'
import './IntroScreen.css';
import PropTypes from 'prop-types';

const IntroScreen = ( props ) => {

  return (
    <div className="intro-container">
      <div className="title-container" >
        <h1 className="sw-logo" >swapi</h1>
          <h1 className="sw-logo-film" >{props.title}</h1>
        <h1 className="sw-logo" >box</h1>
      </div>
      {/* <img alt="logo" src={logo} className="intro-logo" /> */}
      <div className="intro-screen" id="board" >
        <div id="content">
          <p className="crawl-head" id="subtitle">{props.title}</p>
          <p className="crawl-head" id="title">{props.year}</p>
          <div className="crawl-text">
              {props.crawl}
          </div>
        </div>
      </div>
      <button className="intro-arrow" >
        <img alt="" src={rebelClear} className="rebellion-button" />
      </button>
    </div>
  )
}

IntroScreen.propTypes = {
  crawl: PropTypes.string,
  title: PropTypes.string,
  year: PropTypes.string
}

export default IntroScreen;