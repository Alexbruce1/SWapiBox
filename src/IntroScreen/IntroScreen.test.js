import React from 'react';
import { configure, shallow } from 'enzyme';

import IntroScreen from './IntroScreen';

describe('IntroScreen', () => {
  it('Should match the snapshot', () => {
    const mockcrawl = 'A long time ago'
    const mocktitle = 'Return of the Sithe'
    const mockyear =  '1992'

    const wrapper = shallow(<IntroScreen  crawl={mockcrawl} title={mocktitle} year={mockyear} />)
    expect(wrapper).toMatchSnapshot()
  })


})