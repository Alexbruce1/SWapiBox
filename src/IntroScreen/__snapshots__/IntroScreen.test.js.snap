// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`IntroScreen Should match the snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <IntroScreen
    crawl="A long time ago"
    title="Return of the Sithe"
    year="1992"
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <div
          className="title-container"
        >
          <h1
            className="sw-logo"
          >
            swapi
          </h1>
          <h1
            className="sw-logo-film"
          >
            Return of the Sithe
          </h1>
          <h1
            className="sw-logo"
          >
            box
          </h1>
        </div>,
        <div
          className="intro-screen"
          id="board"
        >
          <div
            id="content"
          >
            <p
              className="crawl-head"
              id="subtitle"
            >
              Return of the Sithe
            </p>
            <p
              className="crawl-head"
              id="title"
            >
              1992
            </p>
            <div
              className="crawl-text"
            >
              A long time ago
            </div>
          </div>
        </div>,
        <button
          className="intro-arrow"
        >
          <img
            alt=""
            className="rebellion-button"
            src="rebel-yellow.svg"
          />
        </button>,
      ],
      "className": "intro-container",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <h1
              className="sw-logo"
            >
              swapi
            </h1>,
            <h1
              className="sw-logo-film"
            >
              Return of the Sithe
            </h1>,
            <h1
              className="sw-logo"
            >
              box
            </h1>,
          ],
          "className": "title-container",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "swapi",
              "className": "sw-logo",
            },
            "ref": null,
            "rendered": "swapi",
            "type": "h1",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Return of the Sithe",
              "className": "sw-logo-film",
            },
            "ref": null,
            "rendered": "Return of the Sithe",
            "type": "h1",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "box",
              "className": "sw-logo",
            },
            "ref": null,
            "rendered": "box",
            "type": "h1",
          },
        ],
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <div
            id="content"
          >
            <p
              className="crawl-head"
              id="subtitle"
            >
              Return of the Sithe
            </p>
            <p
              className="crawl-head"
              id="title"
            >
              1992
            </p>
            <div
              className="crawl-text"
            >
              A long time ago
            </div>
          </div>,
          "className": "intro-screen",
          "id": "board",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <p
                className="crawl-head"
                id="subtitle"
              >
                Return of the Sithe
              </p>,
              <p
                className="crawl-head"
                id="title"
              >
                1992
              </p>,
              <div
                className="crawl-text"
              >
                A long time ago
              </div>,
            ],
            "id": "content",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Return of the Sithe",
                "className": "crawl-head",
                "id": "subtitle",
              },
              "ref": null,
              "rendered": "Return of the Sithe",
              "type": "p",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "1992",
                "className": "crawl-head",
                "id": "title",
              },
              "ref": null,
              "rendered": "1992",
              "type": "p",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "A long time ago",
                "className": "crawl-text",
              },
              "ref": null,
              "rendered": "A long time ago",
              "type": "div",
            },
          ],
          "type": "div",
        },
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <img
            alt=""
            className="rebellion-button"
            src="rebel-yellow.svg"
          />,
          "className": "intro-arrow",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "alt": "",
            "className": "rebellion-button",
            "src": "rebel-yellow.svg",
          },
          "ref": null,
          "rendered": null,
          "type": "img",
        },
        "type": "button",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <div
            className="title-container"
          >
            <h1
              className="sw-logo"
            >
              swapi
            </h1>
            <h1
              className="sw-logo-film"
            >
              Return of the Sithe
            </h1>
            <h1
              className="sw-logo"
            >
              box
            </h1>
          </div>,
          <div
            className="intro-screen"
            id="board"
          >
            <div
              id="content"
            >
              <p
                className="crawl-head"
                id="subtitle"
              >
                Return of the Sithe
              </p>
              <p
                className="crawl-head"
                id="title"
              >
                1992
              </p>
              <div
                className="crawl-text"
              >
                A long time ago
              </div>
            </div>
          </div>,
          <button
            className="intro-arrow"
          >
            <img
              alt=""
              className="rebellion-button"
              src="rebel-yellow.svg"
            />
          </button>,
        ],
        "className": "intro-container",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <h1
                className="sw-logo"
              >
                swapi
              </h1>,
              <h1
                className="sw-logo-film"
              >
                Return of the Sithe
              </h1>,
              <h1
                className="sw-logo"
              >
                box
              </h1>,
            ],
            "className": "title-container",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "swapi",
                "className": "sw-logo",
              },
              "ref": null,
              "rendered": "swapi",
              "type": "h1",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Return of the Sithe",
                "className": "sw-logo-film",
              },
              "ref": null,
              "rendered": "Return of the Sithe",
              "type": "h1",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "box",
                "className": "sw-logo",
              },
              "ref": null,
              "rendered": "box",
              "type": "h1",
            },
          ],
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <div
              id="content"
            >
              <p
                className="crawl-head"
                id="subtitle"
              >
                Return of the Sithe
              </p>
              <p
                className="crawl-head"
                id="title"
              >
                1992
              </p>
              <div
                className="crawl-text"
              >
                A long time ago
              </div>
            </div>,
            "className": "intro-screen",
            "id": "board",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <p
                  className="crawl-head"
                  id="subtitle"
                >
                  Return of the Sithe
                </p>,
                <p
                  className="crawl-head"
                  id="title"
                >
                  1992
                </p>,
                <div
                  className="crawl-text"
                >
                  A long time ago
                </div>,
              ],
              "id": "content",
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "Return of the Sithe",
                  "className": "crawl-head",
                  "id": "subtitle",
                },
                "ref": null,
                "rendered": "Return of the Sithe",
                "type": "p",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "1992",
                  "className": "crawl-head",
                  "id": "title",
                },
                "ref": null,
                "rendered": "1992",
                "type": "p",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "A long time ago",
                  "className": "crawl-text",
                },
                "ref": null,
                "rendered": "A long time ago",
                "type": "div",
              },
            ],
            "type": "div",
          },
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <img
              alt=""
              className="rebellion-button"
              src="rebel-yellow.svg"
            />,
            "className": "intro-arrow",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "alt": "",
              "className": "rebellion-button",
              "src": "rebel-yellow.svg",
            },
            "ref": null,
            "rendered": null,
            "type": "img",
          },
          "type": "button",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
